### PR TITLE
Make Residents act as list

### PR DESCRIPTION
**Pull request by zbarnard1**
Adds a new dependency that will help us track the position of the Residents for the User to pray for, as well as 2 endpoints for retrieving either a particular Resident by ID or the next resident the User will pray for based on the last Prayer they made.

### CODE REVIEW

File: app/app.rb [hunk 1]
```diff
@@ -5,32 +5,51 @@
 require 'sinatra/activerecord'
 require 'sinatra/cors'
 require_relative './models'
+register Sinatra::Cors # Add this line.
```
Comment (zbarnard1) [inline app/app.rb:8]: I don't remember where this came from, I think I was trying to get the header to work? This should probably leave unless we need it.

File: app/app.rb [hunk 2]
```diff
@@ -5,32 +5,51 @@
...
-  verse = verses.select { |v| v.day == day }
-
-  verse = verses.select { |v| v.day == 1 } if verse.empty?
-  content_type :json
-  {
-    verse: verse && verse.first.scripture,
-    notation: verse && verse.first.notation,
-    version: verse && verse.first.version
-  }.to_json
+  # get the verse of the day
```
Comment (zbarnard1) [inline app/app.rb:84]: Rubocop auto formatted this, I think I just need to remove it.

**Review by ittullos — CHANGES_REQUESTED** [review]
Looks great! Could we scope the name validation of the resident to the user?

File: app/models/resident.rb [hunk 3]
```diff
@@ -1,5 +1,15 @@
...
-  validates_uniqueness_of :name, case_sensitive: false
+  belongs_to :user
+
+  validates_uniqueness_of :name, case_sensitive: false, scope: :user_id
+
+  acts_as_list scope: :user
+
+  def next_resident
+    lower_item || user.residents.limit(1).first
+  end
```
Comment (ittullos) [inline app/models/resident.rb:14]: So I'm guessing this gem is just doing all the work for us here and keeping up with our place in the list as we go? On line 13 we're essentially just saying give me the next resident or give me the first one if nil?

File: spec/support/helpers/api_spec_helpers.rb
Comment (ittullos) [inline spec/support/helpers/api_spec_helpers.rb:1]: This is my first time seeing API spec helpers. Maybe we can chat a bit about what they do and why they are necessary.

**Review by ittullos — COMMENTED** [review]
This all looks great dude! Was a good refresher for me on a few things too!

File: app/models/resident.rb (hunk 3 above)
Comment (zbarnard1) [reply app/models/resident.rb:14]: The gem doesn't really "keep track" of the position, but if we have an instance of a Resident we can navigate down the list using `lower_item`. This returns the next record in the list relative to the instance you called it on.

File: spec/support/helpers/api_spec_helpers.rb
Comment (zbarnard1) [reply spec/support/helpers/api_spec_helpers.rb:1]: Yeah we can dig into it maybe this week, but I'll comment a bit here for now.
The spec helpers are really just a Ruby module. You can include a module in a class and it makes the code in the module available to that class. The idea is that you can separate your re-usable code into modules that can be included where they are needed and it keeps your codebase DRY.
The code in the APISpecHelpers is meant to be used by every request spec, so making it a module makes sharing this code easy. It's not required to use them, but it makes it much easier to parse the response and check it for the data that's important to the test.

Comment (ittullos) [reply spec/support/helpers/api_spec_helpers.rb:1]: Awesome! Makes sense!

**Pull request merged**